modelon/impact/client/entities/workspace.py

```
@@ -26,6 +28,42 @@
 ]
 
 
+class WorkspaceDefinition:
+    def __init__(self, data: Dict[str, Any]):
+        self.data = data
+
+    def to_dict(self) -> Dict[str, Any]:
+        return self.data
+
+    def to_file(self, path: Optional[str] = None) -> str:
+        if path is None:
+            path = os.path.join(tempfile.gettempdir(), "impact-downloads")
+        os.makedirs(path, exist_ok=True)
+        definition_path = os.path.join(path, self.data['definition']['name'] + ".json")
+        with open(definition_path, "w", encoding='utf-8') as f:
+            json.dump(self.data, f, indent=4)
+        return definition_path
+
+    def with_selected_matchings(self, entries):
+        selected_matchings = {"selectedMatchings": {"entries": entries}}
+        return WorkspaceDefinition({**self.data, **selected_matchings})
+
+    @classmethod
+    def from_file(cls, path: str):
+        with open(path) as json_file:
+            data = json.load(json_file)
+        return cls(data)
+
+    def get_vcs_uri_from_reference_id(self, reference_id: str):
+        projects = (
+            self.data["definition"]["projects"]
+            + self.data["definition"]["dependencies"]
+        )
+        for project in projects:
+            if project["reference"]["id"] == reference_id:
```

What happens if there is no match, should we raise an error?